File: presentation/src/test/java/io/tripled/social/client/presentation/controller/ReadControllerTest.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.test.SpiedReadMessagesUseCase;
import io.tripled.social.client.presentation.TestOutput;
import io.tripled.social.client.presentation.request.ReadCliRequest;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;

public class ReadControllerTest {

  private ReadController readController;
  private SpiedReadMessagesUseCase service;

  @Before
  public void setUp() {
    service = new SpiedReadMessagesUseCase();
    readController = new ReadController(service);
  }

  @Test
  public void executesCommand() {
    ReadCliRequest request = new ReadCliRequest("Alice");

    readController.execute(request, new TestOutput());

    assertThat(service.userName, equalTo("Alice"));
  }

}

File: src/test/kotlin/io/tripled/social/client/infrastructure/cli/controller/ReadControllerTest.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.usecase.spies.ReadMessagesSpy
import io.tripled.social.client.infrastructure.cli.TestOutput
import io.tripled.social.client.infrastructure.cli.request.ReadCliRequest
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class ReadControllerTest {
    private var readController: ReadController? = null
    private var service: ReadMessagesSpy? = null

    @BeforeEach
    fun setUp() {
        service = ReadMessagesSpy()
        readController = ReadController(service!!)
    }

    @Test
    fun executesCommand() {
        val request = ReadCliRequest("Alice")

        readController!!.execute(request, TestOutput())

        MatcherAssert.assertThat<String?>(service!!.userName, Matchers.equalTo<String?>("Alice"))
    }
}